File: lambdas/orden/ListarOrdenes.py
import boto3
import json

dynamodb = boto3.resource('dynamodb')
table = dynamodb.Table('pf_ordenes')

def lambda_handler(event, context):
    try:
        user_id = event['query']['user_id']
        order_id = event['query']['order_id']

        if not user_id or not order_id:
            return {
                'statusCode': 400,
                'body': {'message': 'user_id y order_id son requeridos.'}
            }

        response = table.query(
            IndexName='user_id-order_id-index',
            KeyConditionExpression=Key('user_id').eq(user_id) & Key('order_id').eq(order_id)
        )

        return {
            'statusCode': 200,
            'body': {
                'message': 'Órdenes listadas exitosamente',
                'orders': response.get('Items', [])
            }
        }

    except Exception as e:
        return {
            'statusCode': 500,
            'body': {'message': f'Error al listar órdenes: {str(e)}'}
        }

File: lambdas/orden/ListarOrdenes.js
const { DynamoDBClient } = require("@aws-sdk/client-dynamodb");
const { DynamoDBDocumentClient, QueryCommand } = require("@aws-sdk/lib-dynamodb");

const client = new DynamoDBClient({});
const dynamo = DynamoDBDocumentClient.from(client);

exports.lambda_handler = async (event) => {
    try {
        const tenant_id = event.query.tenant_id;
        const user_id = event.query.user_id;

        if (!tenant_id || !user_id) {
            return {
                statusCode: 400,
                body: { message: "tenant_id y user_id son obligatorios" },
            };
        }

        const params = {
            TableName: "pf_orders",
            IndexName: "tenant_id-user_id-index",
            KeyConditionExpression: "tenant_id = :tenant_id AND user_id = :user_id",
            ExpressionAttributeValues: {
                ":tenant_id": tenant_id,
                ":user_id": user_id,
            },
        };

        const result = await dynamo.send(new QueryCommand(params));

        if (result.Items && result.Items.length > 0) {
            return {
                statusCode: 200,
                body: result.Items,
            };
        } else {
            return {
                statusCode: 404,
                body: { 
                    message: "No se encontraron órdenes para este usuario" 
                }
            };
        }
    } catch (error) {
        console.error("Error al listar órdenes:", error);
        return {
            statusCode: 500,
            body: { message: `Error al listar órdenes: ${error.message}` }
        };
    }
};
